### PR TITLE
add ignore index for sigmoid cross entropy with logits op, test=develop

### PR DESCRIPTION
**Pull request by jerrywgz**

### CODE REVIEW

**Review by qingqing01 — DISMISSED** [review]

**Review by panyx0718 — DISMISSED** [review]
In a follow up PR, can you create a constant called kIgnoreIndex=100 and use this constant for both python and C++? So we don't get magic -100 in many places.

**Review by qingqing01 — APPROVED** [review]

**Review by panyx0718 — APPROVED** [review]
LG API
